osmdroid-android/src/main/java/org/osmdroid/util/RectL.java

```diff
@@ -316,4 +316,40 @@ public static long getRotatedY(final long pX, final long pY,
                                    final double pCos, final double pSin) {
         return pCenterY + Math.round((pX - pCenterX) * pSin + (pY - pCenterY) * pCos);
     }
+
+    /**
+     * @since 6.0.2
+     */
+    public void offset(final long pDx, final long pDy) {
+        left += pDx;
+        top += pDy;
+        right += pDx;
+        bottom += pDy;
+    }
+
+    /**
+     * @since 6.0.2
+     */
+    public void union(final long pLeft, final long pTop, final long pRight, final long pBottom) {
+        if ((pLeft < pRight) && (pTop < pBottom)) {
+            if ((left < right) && (top < bottom)) {
+                if (left > pLeft) left = pLeft;
```

Comment:
You could also write it like `top = Math.max(top, pTop);`

Reply:
Just a simple copy of the `Rect.union` code. And I guess the code will be identically easy to read and quick to run in the `Rect` version and in your suggestion.

Reply:
Sure, that's why it's  a comment and not a task:)